pkg/controller/controller_instance.go

```diff
@@ -1522,49 +1485,67 @@ func (c *controller) pollServiceInstance(instance *v1beta1.ServiceInstance) erro
 		return c.finishPollingServiceInstance(instance)
 	default:
 		glog.Warning(pcb.Messagef("Got invalid state in LastOperationResponse: %q", response.State))
-		if !time.Now().Before(instance.Status.OperationStartTime.Time.Add(c.reconciliationRetryDuration)) {
-			clone, err := api.Scheme.DeepCopy(instance)
-			if err != nil {
-				return err
-			}
-			toUpdate := clone.(*v1beta1.ServiceInstance)
-			s := "Stopping reconciliation retries on ServiceInstance because too much time has elapsed"
-			glog.Info(pcb.Message(s))
-			c.recorder.Event(instance, corev1.EventTypeWarning, errorReconciliationRetryTimeoutReason, s)
+		if c.reconciliationTimeExpired(instance) {
+			return c.reconciliationTimeExpiredFinishPollingServiceInstance(instance, mitigatingOrphan, provisioning, deleting)
+		}
+		return fmt.Errorf(`Got invalid state in LastOperationResponse: %q`, response.State)
+	}
+	return nil
+}
 
-			if mitigatingOrphan {
-				setServiceInstanceCondition(
-					toUpdate,
-					v1beta1.ServiceInstanceConditionReady,
-					v1beta1.ConditionUnknown,
-					errorOrphanMitigationFailedReason,
-					"Orphan mitigation failed: "+s)
-			} else {
-				setServiceInstanceCondition(toUpdate,
-					v1beta1.ServiceInstanceConditionFailed,
-					v1beta1.ConditionTrue,
-					errorReconciliationRetryTimeoutReason,
-					s)
-			}
+// reconciliationTimeExpired tests if the current Operation State time has
+// elapsed the reconciliationRetryDuration time period
+func (c *controller) reconciliationTimeExpired(instance *v1beta1.ServiceInstance) bool {
+	if !time.Now().Before(instance.Status.OperationStartTime.Time.Add(c.reconciliationRetryDuration)) {
+		return true
```

Comment:
I'd just return the value of `time.Now().yadayada...` directly.

Reply:
Talked offline, I prefer the more verbose way in the sake for readability. One change could be to remove the `!` and flip the `true` and `false` to allow readers to parse the `.Before` logic.

Perhaps as a follow-up when the tests are written.